lib/Doctrine/DBAL/Driver/OCI8/OCI8Connection.php

```diff
@@ -153,7 +147,7 @@ public function lastInsertId($name = null)
             throw new OCI8Exception('lastInsertId failed: Query was executed but no result was returned.');
         }
 
-        return (int) $result;
+        return (string) $result;
```

Comment:
Is casting to string needed? Most likely, the previously existing casting was needed since the driver returns numbers as strings which is exactly what we need.

Reply:
That's hard to know without activating `strict_types` and hoping that the test suite is comprehensive enough! I would be surprised if the database itself didn't return a sequence value as an integer, however I don't know how OCI8 transports the data. Hopefully it's not like PDO which casts as `string` or not depending on a seemingly unrelated setting (namely ATTR_EMULATE_PREPARES). In doubt, does the `(string)` cast hurt?

Reply:
> That's hard to know without activating `strict_types`

Not really:
```php
$stmt = oci_parse($conn, 'SELECT 1 FROM DUAL');
oci_execute($stmt);
$row = oci_fetch_assoc($stmt);

var_dump($row);

/*
array(1) {
  [1] =>
  string(1) "1"
}
*/
```
It doesn't hurt but it's not needed and looks confusing. Most DB drivers return numbers (integer and float) as strings. The reason is that sometimes, the precision a DB platforms provides (e.g. `BIGINT` on MySQL) is higher than the PHP can handle on the current OS platform (e.g. x86). That's why drivers return number as strings and let the application do the conversion if needed.

Reply:
Fair enough. I'm never this confident as, as I said above, at least PDO can return MySQL's integers as `int` depending on a configuration setting (probably only on x64, and/or only for ints < 2^31, I haven't tested). I hope there is no such thing in OCI8; anyway, once strict_types will be enabled everywhere, and if our tests are extensive enough, we should spot any such issue later on.

Anyway, I've removed the type cast.